tests/nlu/training_data/test_markdown.py

```diff
@@ -0,0 +1,271 @@
+from typing import Optional, Text, Dict, Any, List
+
+import pytest
+
+from rasa.nlu import load_data
+from rasa.nlu.extractors.crf_entity_extractor import CRFEntityExtractor
+from rasa.nlu.extractors.duckling_http_extractor import DucklingHTTPExtractor
+from rasa.nlu.extractors.mitie_entity_extractor import MitieEntityExtractor
+from rasa.nlu.extractors.spacy_entity_extractor import SpacyEntityExtractor
+from rasa.nlu.training_data.formats import RasaReader
+from rasa.nlu.training_data.formats import MarkdownReader, MarkdownWriter
+
+
+@pytest.mark.parametrize(
+    "example, expected_entities, expected_text",
+    [
+        ("i'm looking for a place to eat", None, "i'm looking for a place to eat"),
+        (
+            "i'm looking for a place in the [north](loc-direction) of town",
+            [{"start": 31, "end": 36, "value": "north", "entity": "loc-direction"}],
+            "i'm looking for a place in the north of town",
+        ),
+        (
+            "show me [chines](cuisine:chinese) restaurants",
+            [{"start": 8, "end": 14, "value": "chinese", "entity": "cuisine"}],
+            "show me chines restaurants",
+        ),
+        (
+            'show me [italian]{"entity": "cuisine", "value": "22_ab-34*3.A:43er*+?df"} '
+            "restaurants",
+            [
+                {
+                    "start": 8,
+                    "end": 15,
+                    "value": "22_ab-34*3.A:43er*+?df",
+                    "entity": "cuisine",
+                }
+            ],
+            "show me italian restaurants",
+        ),
+        ("Do you know {ABC} club?", None, "Do you know {ABC} club?"),
+        (
+            "show me [chines](22_ab-34*3.A:43er*+?df) restaurants",
+            [{"start": 8, "end": 14, "value": "43er*+?df", "entity": "22_ab-34*3.A"}],
+            "show me chines restaurants",
+        ),
+        (
+            'I want to fly from [Berlin]{"entity": "city", "role": "to"} to [LA]{'
+            '"entity": "city", "role": "from", "value": "Los Angeles"}',
+            [
+                {
+                    "start": 19,
+                    "end": 25,
+                    "value": "Berlin",
+                    "entity": "city",
+                    "role": "to",
+                },
+                {
+                    "start": 29,
+                    "end": 31,
+                    "value": "Los Angeles",
+                    "entity": "city",
+                    "role": "from",
+                },
+            ],
+            "I want to fly from Berlin to LA",
+        ),
+        (
+            'I want to fly from [Berlin](city) to [LA]{"entity": "city", "role": '
```

Comment:
what happens if the json is invalid? e.g. `I want to fly from [Berlin](city) to [LA]{"entity: "city"}` (missing closing quote). It would be great if the user get's a helpful error message and just a `something went wrong`)